cdk/lambda/upload_test_artifacts/upload_test_artifacts.py

```diff
@@ -0,0 +1,50 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import boto3
+
+CFN_STACK_TAG_KEY = 'aws:cloudformation:stack-id'
+DELETE_STATUS = 'DELETE_IN_PROGRESS'
+SOURCE_BUCKET_EXPORT_NAME = 'MultiplayerTestScalerArtifactBucketName'
+DEFAULT_DESTINATION_BUCKET_EXPORT_NAME = 'O3deMetricsUploadBucket'
+
+def handler(event, context):
+    stack_id = event['resources'][0]
+    new_stack_status = event['detail']['status-details']['status']
+    
+    print(f'Status for stack {stack_id} has changed to: {new_stack_status}')
+    if new_stack_status != DELETE_STATUS:
+        print(f'Status change for stack is not {DELETE_STATUS}, NoOp.')
+        return {
+            'statusCode': 200,
+        }
+ 
+    # get artifact and metrics bucket names
+    cfn = boto3.client('cloudformation')
+    response = cfn.list_exports()
+ 
+    source_bucket = ''
+    destination_bucket = ''
+    for export in response['Exports']:
+        if export['Name'] == SOURCE_BUCKET_EXPORT_NAME:
+            source_bucket = export['Value']
+            print(f'source_bucket is: {source_bucket}')
+        if export['Name'] == DEFAULT_DESTINATION_BUCKET_EXPORT_NAME:
+            destination_bucket = export['Value']
+            print(f'destination_bucket is: {destination_bucket}')
+ 
+    if destination_bucket == '' or source_bucket == '':
+        raise RuntimeError('Required bucket missing! No action will be taken.')
+ 
+    # upload bucket content with unique key
+    s3_client = boto3.client('s3')
+    list_response = s3_client.list_objects(Bucket=source_bucket)
```

Comment:
nit: recommendation is to use https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/list_objects_v2.html over list_objects.

Also paginated API so you could miss results!